api/src/main/java/io/opentelemetry/api/trace/TraceState.java

```diff
@@ -24,21 +21,40 @@
  *
  * <p>Value is opaque string up to 256 characters printable ASCII RFC0020 characters (i.e., the
  * range 0x20 to 0x7E) except comma , and =.
+ *
+ * <p>Implementations of this interface *must* be immutable and have well-defined value-based
+ * equals/hashCode implementations. If an implementation does not strictly conform to these
+ * requirements, behavior of the OpenTelemetry APIs and default SDK cannot be guaranteed.
+ *
+ * <p>Implementations of this interface that do not conform to the W3C specification risk
+ * incompatibility with W3C-compatible implementations.
+ *
+ * <p>For these reasons, it is strongly suggested that you use the implementation that is provided
+ * here via the {@link TraceStateBuilder}.
  */
 @Immutable
-@AutoValue
-public abstract class TraceState {
-  private static final TraceState DEFAULT = TraceState.builder().build();
+public interface TraceState {
+
+  TraceState DEFAULT = new TraceStateBuilder().build();
```

Comment:
Move this to a package private field in `ArrayBasedTraceState` so it's not public

Reply:
I tried this, but it ended up breaking with a class loading issue like the other one. :(

Reply:
What if it's in the builder? I think one of the two is supposed to work ok (or otherwise I guess another package private class)

Reply:
If I were to put this into a separate class, what would I call it? TraceStateDefaultHolder?

Reply:
I called it `DefaultTraceState`. Seems like an obvious enough name. :)